core/math/vector2.cpp

```diff
@@ -47,7 +47,7 @@ real_t Vector2::length_squared() const {
 }
 
 void Vector2::normalize() {
-	real_t l = x * x + y * y;
+	real_t l = length_squared();
```

Comment:
`normalize` is a fairly critical method in the core engine, and we want it to have high performance. Do we know if this function gets inlined? It may be better to just not change this. The changes outside of `vector2.cpp` look good.

Reply:
I did it because Vector3 does it.
https://github.com/godotengine/godot/blob/c6f6c05a521c407870cfa3a979044fdc0e483ff0/core/math/vector3.h#L420-L421

But, I just realized they are force_inline'd.
https://github.com/godotengine/godot/blob/c6f6c05a521c407870cfa3a979044fdc0e483ff0/core/math/vector3.h#L85-L87

But Vector2's aren't force_inline'd.
https://github.com/godotengine/godot/blob/c6f6c05a521c407870cfa3a979044fdc0e483ff0/core/math/vector2.h#L81-L83

So,it would probably be a good idea to force_inline them on separate PR.
As of now, I reverted the above method as you mentioned.